common/base-api.js

```diff
@@ -253,16 +269,20 @@ class BaseApi {
   keyExistsSync(key) {
     if (this.persistent) { // local storage
       if (key in localStorage) {
-        const obj = JSON.parse(localStorage.getItem(key));
-        if (this._timeout === null && obj.kt === null) {
-          return true;
-        } else {
-          if (this._isCacheStale(obj)) {
-            this.removeKeySync();
-            return false;
-          } else {
+        try {
+          const obj = JSON.parse(localStorage.getItem(key));
+          if (this._timeout === null && obj.kt === null) {
             return true;
+          } else {
+            if (this._isCacheStale(obj)) {
+              this.removeKeySync();
+              return false;
+            } else {
+              return true;
+            }
           }
+        } catch (e) {
+          return false;
```

Comment:
We're inside the "key in localStorage" block. Thus the key exists. The value is not JSON, but nonetheless the key does exist.

Please update to return true in this catch block.

Reply:
My reasoning in this case was that, if the key is not valid JSON it's not being managed by this tool, so returning false seemed the logic thing.

Anyway it can work the other way too. So I'm going to update the review shortly.